app/main/views/questionnaire.py

```diff
@@ -215,8 +216,12 @@ def submit_answers(eq_id, form_type, collection_id):
     is_valid, invalid_location = q_manager.validate_all_answers()
 
     if is_valid:
+        metadata = get_metadata(current_user)
+        navigator = Navigator(g.schema_json, metadata, get_answer_store(current_user))
```

Comment:
For conciseness, would it be better to assign answer_store to a var and pass to each method?